File: day11.cpp
#include <bits/stdc++.h>
using namespace std;
#define INPUT(var) for(string var; getline(cin, var), var!="";)
#define FOR(i, s, e) for (int i=s; i<e; ++i)

void solve(int value) {
	vector<string> grid;
	INPUT(line) grid.push_back(line);

	int n = size(grid), m = size(grid[0]);
	vector<bool> row(n, true), col(m, true);
	vector<pair<int, int>> arr;
	FOR(r, 0, n)
	FOR(c, 0, m) {
		row[r] = row[r] && grid[r][c] == '.';
		col[c] = col[c] && grid[r][c] == '.';
		if (grid[r][c] == '#') arr.push_back({ r, c });
	}
	
	long long ans = 0;
	FOR(i, 0, arr.size())
	FOR(j, i+1, arr.size()) {
		auto [r, c] = arr[i];
		auto [R, C] = arr[j];
		if (r > R) swap(r, R);
		if (c > C) swap(c, C);
		FOR(k, r, R) {
			if (row[k]) ans += value;
			else ++ans;
		}
		FOR(k, c, C) {
			if (col[k]) ans += value;
			else ++ans;
		}
	}
	cout << ans << '\n';
}

void main() {
	solve(2);
	solve(1000000);
}

File: day11.py
def solve(value):
	grid = []
	while line := input():
		grid.append(line)

	n, m = len(grid), len(grid[0])
	row = [True for _ in range(n)]
	col = [True for _ in range(m)]
	arr = []
	for r in range(n):
		for c in range(m):
			row[r] &= grid[r][c] == "."
			col[c] &= grid[r][c] == "."
			if grid[r][c] == "#":
				arr.append([r, c])
	
	ans = 0
	for i in range(len(arr)):
		for j in range(i+1, len(arr)):
			r, c = arr[i]
			R, C = arr[j]
			
			if r > R: r, R = R, r
			if c > C: c, C = C, c

			for k in range(r, R):
				if row[k]: ans += value
				else: ans += 1
			for k in range(c, C):
				if col[k]: ans += value
				else: ans += 1
	print(ans)

solve(2)
solve(1000000)